nav2-usability/scenario-code/sign-in-routing/router.dart

```diff
@@ -0,0 +1,360 @@
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+class Credentials {
+  final String username;
+  final String password;
+
+  Credentials(this.username, this.password);
+}
+
+abstract class Authentication {
+  Future<bool> isSignedIn();
+
+  Future<void> signOut();
+
+  Future<bool> signIn(String username, String password);
+}
+
+class MockAuthentication implements Authentication {
```

Comment:
Why don't we implement the authentication service with ChangeNotifier?

Reply:
It's meant to represent an API like [GoogleSignIn](https://pub.dev/documentation/google_sign_in/latest/google_sign_in/GoogleSignIn-class.html)

Reply:
ah, I see you use AppState to notify also auth status. So don't need a separated ChangeNotifier.

Reply:
Yeah, the AppState notifies its listeners on behalf of the Authentication object changes.